### PR TITLE
Expose deadline argument for DAG

### PR DESCRIPTION
**Pull request by ktsitsi**
This PR:

- Exposes the deadline argument for DAG timeouts to the bioimg ingestion API.

### CODE REVIEW

File: src/tiledb/cloud/bioimg/ingestion.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def ingest(
     converter: Optional[str] = None,
     output_ext: str = "",
     tile_scale: int = 128,
+    deadline: Optional[int] = None,
```
[the remark this answers is not in the document]
Comment (ktsitsi) [reply src/tiledb/cloud/bioimg/ingestion.py:48]: Sure!

**Review by sgillies — APPROVED** [review]
@ktsitsi do we need to add any documentation about the expected state of an ingest when it times out? Or whether a user needs to take any cleanup actions?

**Comment by spencerseale** [comment]
@sgillies no cleanup will be needed thanks to validation steps @ktsitsi has put in place. More docs the better

**Comment by spencerseale** [comment]
@ktsitsi can we include the default rather than None? I know None will result in using REST's timeout which is 24 hours, but `timeout: int = 86400` would make it more obvious. We could also say in the doc str the default is 24 hours, but that would need to stay in sync with any potential changes in the REST config. 

Users will want to know what the default timeout is I think

**Comment by sgillies** [comment]
Failures are unrelated (and addressed in main). Merging.

**Pull request merged**